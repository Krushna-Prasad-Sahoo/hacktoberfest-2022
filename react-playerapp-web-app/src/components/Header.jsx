import React from 'react'
import { NavLink } from 'react-router-dom'
export default function Header(props) {
    return (
        <nav class="navbar navbar-expand-lg navbar-dark bg-dark">
            <div>
                <>
                    <ul>
                        <li><NavLink to="/addplayer">Addplayer</NavLink></li>
                        <li><NavLink to="/showplayer">Showplayer</NavLink></li>
                    </ul>
                </>
            </div>

        </nav>
    )
}
