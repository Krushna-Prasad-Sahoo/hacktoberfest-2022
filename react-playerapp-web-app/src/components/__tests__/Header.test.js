 
import Header from '../Header'
import{Link} from 'react-router-dom'
import toJson from 'enzyme-to-json';
import Enzyme, { shallow, render, mount } from 'enzyme';
describe("Header testing",()=>{
  const wrapper = shallow(<Header text='Playerapp' />) 
it('renders correctly Header', () => {   
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should have nav',()=>{
    expect(wrapper.contains(<Link class="navbar-brand" to="/">Playerapp</Link>)).toBeTruthy();
  })
  it('includes link AddPlayer', () => {                                       
    // const wrapper = shallow(<MemoryRouter><Home /></MemoryRouter>);
    // expect(wrapper.find('Link')[1].prop('to')).toEqual('/Showlayer');
    expect(wrapper.find(Link).last().props().to).toEqual('/ShowPlayer');
   });
   it('includes link AddPlayer', () => {                                       
    // const wrapper = shallow(<MemoryRouter><Home /></MemoryRouter>);
    // expect(wrapper.find('Link')[1].prop('to')).toEqual('/AddPlayer');
    expect(wrapper.find(Link).at(1).props().to).toEqual('/AddPlayer');
   });
   it('includes link AddPlayer', () => {                                       
    // const wrapper = shallow(<MemoryRouter><Home /></MemoryRouter>);
    // expect(wrapper.find('Link')[1].prop('to')).toEqual('/');
    expect(wrapper.find(Link).at(0).props().to).toEqual('/');
   });
})
