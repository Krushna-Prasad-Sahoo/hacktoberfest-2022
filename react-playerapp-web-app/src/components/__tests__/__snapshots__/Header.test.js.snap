// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header testing renders correctly Header 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <Link
    class="navbar-brand"
    to="/"
  >
    Playerapp
  </Link>
  <ul
    class="navbar-nav me-auto mb-2 mb-lg-0"
  >
    <li
      class="nav-item"
    >
      <Link
        aria-current="page"
        class="nav-link active"
        to="/AddPlayer"
      >
        Add Player
      </Link>
    </li>
    <li
      class="nav-item"
    >
      <Link
        aria-current="page"
        class="nav-link active"
        to="/ShowPlayer"
      >
        Show Player
      </Link>
    </li>
  </ul>
</nav>
`;
